src/components/manage/Sidebar/SidebarPopup.jsx

```diff
@@ -2,26 +2,43 @@ import React from 'react';
 import { Portal } from 'react-portal';
 import { CSSTransition } from 'react-transition-group';
 import PropTypes from 'prop-types';
-import { doesNodeContainClick } from 'semantic-ui-react/dist/commonjs/lib';
 
 const DEFAULT_TIMEOUT = 500;
 
 const SidebarPopup = (props) => {
   const { children, open, onClose, overlay } = props;
 
-  const asideElement = React.createRef();
-
   const handleClickOutside = (e) => {
-    if (asideElement && doesNodeContainClick(asideElement.current, e)) return;
-    onClose();
+    let sidebarContainer = e?.target?.closest('#sidebar-popup-container');
+    if (open && !sidebarContainer) {
+      onClose();
+    }
+  };
+
+  const handleESCPress = (e) => {
+    if (e.code === 'Escape') onClose();
   };
 
   React.useEffect(() => {
-    document.addEventListener('mousedown', handleClickOutside, false);
-    return () => {
-      document.removeEventListener('mousedown', handleClickOutside, false);
-    };
-  });
+    if (open) {
+      document.addEventListener('click', handleClickOutside, false);
+    } else {
+      document.removeEventListener('click', handleClickOutside, false);
+    }
+    return () =>
+      document.removeEventListener('click', handleClickOutside, false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open]);
```

Comment:
I would consider refactoring the ``handleClickOutside`` function out of the component like ``function handleClickOutside(e, open);`` and fix the eslint comment

Reply:
I also considered that, but even defining handleClickOutside as `function` eslint was expecting to have it as dep 🤷

Reply:
Yes but IIRC that would be fine to add since it won't be redefined on every component render.

Reply:
uh, yeah right

Reply:
I need a function to pass to addEventListener and it would have not the firm of a listener function, so I should have a function which returns the listener and it would be more complicated since the ilstener is recognized by reference.
I'm not sure the performances will benefit more than readability would be worse.
IDK if I'm explaining this.

Reply:
Yeah my point wasn't about performance, it was more about removing eslint fixes, but it's not that much important :)
Anyway yes, you are right, it would be a little more complicated than I imagined.

Reply:
What about useCallback? Than you could pass the result from useCallback as dependency to the useEffect.

Reply:
Yes that's actually the exact solution but for some reason I didn't think about it when I wrote the comment

Reply:
Done in 5f56ae3